### PR TITLE
Syndie Tele Leaves Behind Evidence

### PR DESCRIPTION
**Pull request by 13spacemen**
## About The Pull Request
~~Experimental syndicate teleporter now leaves behind 'bluespace residue' as evidence at start and end locations, contains fingerprints and fibres of the teleporting dude~~
Experimental syndicate teleporter now makes you bleed 10 units and leaves blood splatters at teleport start and end points
## Why It's Good For The Game
blood2
blood

Can find out who has been using syndie teles (as long as they don't clean up the mess), also gives a downside to constant teleportation and makes it so you can't keep it up forever

## Changelog
:cl:
balance: Experimental syndicate teleporter's teleportation is no longer perfect, it will cause the user to bleed a little. Luckily the detective can scan the blood and find out who the big bad is, as long as he's quick.
/:cl:

### CODE REVIEW

**Comment by ArcaneMusic** [comment]
Out of curiosity. How does this work with spraybottles of spacecleaner? Would it travel through the portal and clean both sides?

**Comment by 13spacemen** [comment]
> Out of curiosity. How does this work with spraybottles of spacecleaner? Would it travel through the portal and clean both sides?

Nothing special about the residue, it's basically just there so some evidence exists that someone syndie tele'd into secure areas

**Comment by Jacquerel** [comment]
> Out of curiosity. How does this work with spraybottles of spacecleaner? Would it travel through the portal and clean both sides?

The experimental teleporter doesn't leave a portal, so you can't shoot it behind you to clean up.

**Comment by wraith-54321** [comment]
I would say make it only leave residue at the destination so you can clean it up, or make it only have fibers and not fingerprints.(how would the prints even be getting onto it?)

**Comment by Jacquerel** [comment]
> I would say make it only leave residue at the destination so you can clean it up, or make it only have fibers and not fingerprints.(how would the prints even be getting onto it?)

Not being able to clean it up is the point of the change.

**Comment by Simpleclown69** [comment]
So basically the teleporter would be useless.

**Comment by Zonespace27** [comment]
Good change, gives it its own identity (fast, risky, messy) separate from the bluespace launchpad (slow, risk-free, 0 evidence), instead of one being a near flat-upgrade from the other

**Comment by moocowswag** [comment]
Prints doesnt really make sense but its the closest thing to genetic traces / cells left from a very dangerous and possibly cancer inducing experience from using an experimental knockoff of the hand tele.

**Comment by moocowswag** [comment]
Also reminder this is probably gonna be a cumulative nerf in an attempt to balance this item, this nerf wont fix it entirely because it also grants enough mobility to semi consistently avoid people that already know youre bad.

Tldr: this pr can go with the other nerf prs open, its only an alternative to the removal pr.

File: code/game/objects/items/teleportation.dm [hunk 1]
```diff
@@ -490,6 +492,12 @@
 		victim.Paralyze(6 SECONDS)
 		to_chat(victim, span_warning("[user] teleports into you, knocking you to the floor with the bluespace wave!"))
 
+/obj/item/syndicate_teleporter/proc/make_residues(turf/old_location, turf/new_location, mob/user)
+	var/obj/effect/decal/cleanable/bluespace_residue/entryresidue = new /obj/effect/decal/cleanable/bluespace_residue(old_location)
+	var/obj/effect/decal/cleanable/bluespace_residue/exitresidue = new /obj/effect/decal/cleanable/bluespace_residue(new_location)
```
Comment (san7890) [inline code/game/objects/items/teleportation.dm:497]: snake case, so `entry_residue` and `exit_residue`

File: code/game/objects/items/teleportation.dm [hunk 2]
```diff
@@ -490,6 +492,12 @@
 		victim.Paralyze(6 SECONDS)
 		to_chat(victim, span_warning("[user] teleports into you, knocking you to the floor with the bluespace wave!"))
 
+/obj/item/syndicate_teleporter/proc/make_residues(turf/old_location, turf/new_location, mob/user)
```
Comment (san7890) [inline code/game/objects/items/teleportation.dm:495]: DMDoc this proc.

**Comment by AlamoTurtle** [comment]
I'm not a fan of the DNA stuff, but I do think I wouldn't mind a form of "residue" left behind, though I'm curious if you can make it more obviously Syndicate, like telecrystal dust instead of bluespace residue.

File: code/game/objects/items/teleportation.dm [hunk 3]
```diff
@@ -514,6 +522,15 @@
 /obj/effect/temp_visual/teleport_abductor/syndi_teleporter
 	duration = 5
 
+/obj/effect/decal/cleanable/bluespace_residue
+	name = "bluespace residue"
+	desc = "A pseudoliquid formed when unstable bluespace crystals are subjected to extreme heat."
```
Comment (san7890) [inline code/game/objects/items/teleportation.dm:527]:
```suggestion
	desc = "A liquid-like substance formed when unstable bluespace crystals are subjected to extreme heat."
```

psuedoliquid sounds weird

**Review by san7890 — CHANGES_REQUESTED** [review]

**Comment by ATH1909** [comment]
I'm not a fan of this somehow leaving your fingerprints on the residue. You weren't even touching the residue, how does that even work? I think just the residue existing at all would be fine, no?

At the very least, this should be mentioned in the teleporter's description somehow.

**Comment by BlueMemesauce** [comment]
> Prints doesnt really make sense but its the closest thing to genetic traces / cells left from a very dangerous and possibly cancer inducing experience from using an experimental knockoff of the hand tele.

DNA is much closer to that and already exists in game when you scan blood

**Comment by Wallemations** [comment]
I like the balance change to this overall, but I'm similarly confused over the fingerprint portion. Idk, to me that just feels like it's too easy to trace compared to the fibers.

**Comment by 13spacemen** [comment]
The 'lore' behind it I thought of is that the syndie tele uses unstable bluespace crystals that get melted down and the residue spills out of a hole on the tele and touches your hand

Currently the fingerprints are set to ignore gloves and will be there even if you wear fully covering gloves; that's a bit too easy to scan so I'll change that

**Comment by Livrah** [comment]
Crutch for crutch from bad gamedesign.

**Comment by BlueMemesauce** [comment]
> The 'lore' behind it I thought of is that the syndie tele uses unstable bluespace crystals that get melted down and the residue spills out of a hole on the tele and touches your hand
> 
> Currently the fingerprints are set to ignore gloves and will be there even if you wear fully covering gloves; that's a bit too easy to scan so I'll change that

Is there a way to set it to dna like blood? That would make more sense than fingerprints bypassing gloves

**Comment by 13spacemen** [comment]
> Is there a way to set it to dna like blood? That would make more sense than fingerprints bypassing gloves
not quite sure how DNA works, just from glancing at the code seems like it's tied to blood
anyway right now it no longer bypasses gloves that cover hands

**Comment by ATH1909** [comment]
> The 'lore' behind it I thought of is that the syndie tele uses unstable bluespace crystals that get melted down and the residue spills out of a hole on the tele and touches your hand

... how do fingerprints get left behind on a LIQUID?

blood leaves behind dna evidence because the dna in the blood cells themselves is collected and used as evidence. you're telling me that molten bluespace crystals somehow silently spill out of the device, touch your skin enough to take some skin cells with them (but not enough to damage you, despite being MOLTEN), but don't damage (the dna in) those skin cells with their heat?

**Comment by 13spacemen** [comment]
> ... how do fingerprints get left behind on a LIQUID?

bluespace

**Comment by moocowswag** [comment]
> Crutch for crutch that needed for bad gamedesign.

Sadly its all we are left with because of the opensourceness of the code base so nobody cab make a major coherent change and the only thing people are able to really do on an individual level is small adjustments and features.

But i do think its absurd that we took traitor’s teeth and people argued for removal of the only thing that gave them safety similar to heretic blade shatter.

It seems like the push for longer rounds has done more to hurt the game than to help it.

in order to have longer rounds people individually added

prog tots
Experisci
Heretics
Dynamic
Paychecks/lathe tax

Each of these changes required crutches of their own to work, like rev win no longer ending.

But on the flipside within the last few years weve had changes that made longer rounds more unobtainable with lazy players and early call more likely

Sm became impactful when it does a baby delam. (Anomaly spam)

Nerfs / removals of other power sources.

Nerfs to medbay/removal of nanites which made healing simple damage more annoying.

Nerfs to cryo which made using it long term alittle annoying.

shuttle call now counters almost every round start antag as they are expected to scale over the round so might aswell metagame that shit because you have a personal vendetta to prevent other players from making the most if their antag rounds.

People complained that experimental teleporter is the reason traitors dont interact with the round , uh no it keeps them alive and cooking while they wait 90 minutes to start interacting with the round which will be over by then.

**Comment by ATH1909** [comment]
> bluespace

how is a player who hasn't codedived supposed to know that these piles of sludge somehow have their dna on them, though?

**Review by Twaticus — APPROVED** [review]

**Comment by VladinXXV** [comment]
My main issue here is that the goop is so glaringly obvious to anyone and everyone who sees it. It screams SCAN ME SCAN ME ILL TELL YOU WHO THE VALID IS. I'd personally favor making this more difficult for people other than (ideally) the detective to notice in some way (I dream of giving the det a blacklight but it is probably overkill for this ONE thing.), like only being visible when wearing their special spy glasses or something. Alternatively, make it a more generic mess, something people won't notice so quickly. People don't scream DET COME SCAN VALID JUICE for _every_ puddle of blood they see, it is just too common. But this is a dead giveaway. On that note, blood could be a good substitute appearance. Maybe have it look similar to a fresh blood spill, but call it telecrystal/redspace dust. The observant will notice it is named differently/never dries up.

**Comment by VladinXXV** [comment]
> The experimental teleporter doesn't leave a portal, so you can't shoot it behind you to clean up.

I'm sorry you had to find out like this. The title screen art lied to you. Portal and Portal 2 lied to you. You cannot shoot through portals. You can hardly even throw through them - everything passing through instantly loses all momentum.

**Comment by GuillaumePrata** [comment]
The teleporter as a gambling break-in tool that doesn't leave evidence isn't even that big of a deal.

The issue is more on how easy it is to keep evading conflict using the teleporter, having the name of the valid doesn't change how boring it is to chase a traitor that can just teleport away constantly.

**Review by san7890 — CHANGES_REQUESTED** [review]
Yeah looking through the comments I think I might agree with it being perceived as some sort of "valid juice"... There are two options that I think would be fitting, pick your poison:

A) Make it more red/blood-like. I think that if you retain the same cleanable effect appearance it'll still be distinctive enough to be picked up but it's not going to be like obviously blue... I'm more meh on this one but it could work.

B) Have it "evaporate" over time. It's reactive blue juice right? Maybe some particles will spontaneously disappear over time as it teleports away off to affix itself onto walls or just become entropic dust with everything else. I don't think we have this functionality in the code yet, but it would be nice if pixels from the sprite could slowly and slowly disappear over time... it could also give it a sense of "timeliness" in tracking down a suspect such that you don't get jebaited by an old not-cleaned up bit of residue. I wonder if five minutes for it to evaporate away would be fair?

B is more code-intensive, but I think it would provide a much more atmospheric and cooler downside... it might not even have to evaporate completely but the residue slowly fading away more and more (and maybe after a certain point, the DNA residues become unrecognizable as well) sounds great to me. I'm not the happiest with A but I think it would be acceptable if you are not able/willing to go through with B. Let me know.

**Comment by Pickle-Coding** [comment]
A slightly different version of option A sounds interesting. Some of the blood gets left behind during teleportation, which detectives could scan to see who's blood it would match. It will also make it obvious to the user that they're going to die the more they use it.

**Comment by Jacquerel** [comment]
> B is more code-intensive, but I think it would provide a much more atmospheric and cooler downside... 

You might be able to implement B just by applying the `temporary_atom` component which gives an atom a fixed lifespan and then fades out its alpha over a configurable period of time.

**Comment by castawaynont** [comment]
this doesn't fix the real issues with the teleporter and just doesn't make sense on a ic or player level

**Comment by 13spacemen** [comment]
Ok teleporting now makes you bleed instead

**Comment by castawaynont** [comment]
now THIS is cool

**Comment by DATA-xPUNGED** [comment]
> Ok teleporting now makes you bleed instead

Do you actually lose blood on the process or is a blood puddle of your blood spawned? I'd be interesting if using the syndie teleporter too frequent would make you anemic

**Comment by 13spacemen** [comment]
> Do you actually lose blood on the process or is a blood puddle of your blood spawned? I'd be interesting if using the syndie teleporter too frequent would make you anemic

Its real bloodloss, you probably won't feel it unless you spam teleport every time it goes off cooldown though since its only 10 units. Humans start at 560 units and once you reach ~460 is when you begin taking suffocation damage

File: code/game/objects/items/teleportation.dm [hunk 4]
```diff
@@ -502,6 +511,8 @@
 		<b>Warning:</b> Teleporting into walls will activate a failsafe teleport parallel up to 3 meters, but the user will be ripped apart if it fails to find a safe location.<br>
 		<br>
 		Do not expose the teleporter to electromagnetic pulses. Unwanted malfunctions may occur.
+		<br>
+		Final word of caution: the technology involved is experimental in nature, you may feel lighter after heavy use. Both in weight, and in headedness. This is normal.
```
Comment (san7890) [inline code/game/objects/items/teleportation.dm:515]:
```suggestion
		Final word of caution: the technology involved is experimental in nature. Although many years of research have allowed us to prevent leaving your organs behind, it simply cannot account for all of the liquid in your body.
```

The previous description feels really arcane and I'd rather we just spell it out.

**Review by san7890 — CHANGES_REQUESTED** [review]
good to go after this change is made
